packages/addon-dev/src/rollup-hbs-plugin.ts

```diff
@@ -30,30 +29,34 @@ export default function rollupHbsPlugin({
       }
     },
 
-    load(id: string) {
-      if (hbsFilter(id)) {
-        return getHbsToJSCode(id);
+    async transform(code: string, id: string) {
+      if (!hbsFilter(id)) {
+        return;
       }
       let meta = getMeta(this, id);
-      if (meta) {
-        if (meta?.type === 'template-js') {
-          const hbsFile = id.replace(/\.js$/, '.hbs');
-          return getHbsToJSCode(hbsFile);
-        }
+      if (meta?.type === 'template-only-component-js') {
         return {
-          code: templateOnlyComponent,
+          code: templateOnlyComponent(code),
         };
       }
+      return getHbsToJSCode(code);
     },
   };
 }
 
-const templateOnlyComponent =
-  `import templateOnly from '@ember/component/template-only';\n` +
-  `export default templateOnly();\n`;
+function templateOnlyComponent(hbsCode: string) {
+  const code = hbsCode.replace(/`/g, '\\`').replace(/\$/g, '\\$');
+  return `
+  import templateOnly from '@ember/component/template-only';
+  import { precompileTemplate } from '@ember/template-compilation';
+  import { setComponentTemplate } from '@ember/component';
+  export default setComponentTemplate(precompileTemplate(\`${code}\`), templateOnly());
+  `;
+}
```

Comment:
We were avoiding this kind of inlining on purpose. I would prefer to keep letting rollup do the inlining automatically.

Reply:
I do not see how it can be done when using `transform` in the hbs plugin.

the hbs gets transformed by embroiders babel plugin to
e.g. if we get the file `demo.hbs`
we would have
```js
import { setComponentTemplate } from '@ember/component';
import default as TEMPLATE from './demo.hbs'
export default setComponentTemplate(precompileTemplate(TEMPLATE), templateOnly());
```

but ` './demo.hbs'` is already cached with the transform, so we would get the same thing back

Reply:
i fixed it